tests/cases/fourslash/incrementalParsingDynamicImport1.ts

```diff
@@ -7,11 +7,11 @@
 
 //// var x1 = import("./foo");
 //// x1.then(foo => {
-////    var s: string = foo.bar();    
+////    var s: string = foo.bar();
 //// })
 //// /*1*/
 
-verify.numberOfErrorsInCurrentFile(1);
+verify.numberOfErrorsInCurrentFile(2);

```

Comment:
so what is the extra error that showed up here?

Reply:
I fixed an issue in https://github.com/Microsoft/TypeScript/pull/19452/commits/967a426fe1d175a3cbfa0211df96e020a593507e where we were losing some deferred global diagnostics.

Reply:
When we would call `sortAndDeduplicateDiagnostics` we would sort the diagnostics in place: https://github.com/Microsoft/TypeScript/commit/967a426fe1d175a3cbfa0211df96e020a593507e#diff-66cec844e251a1918deb897eabca206bL1856

After changing that to sort into a new array, I found that we had a few cases where we would modify the sorted array when adding deferred diagnostics. This was the fix: https://github.com/Microsoft/TypeScript/commit/967a426fe1d175a3cbfa0211df96e020a593507e#diff-942ae3cd2a8bbd85ed86a60cd7c43307R2324